cudax/include/cuda/experimental/__device/arch_traits.cuh

```diff
@@ -0,0 +1,401 @@
+  __traits.compute_capability_major             = 8;
+  __traits.compute_capability_minor             = 6;
+  __traits.compute_capability                   = 86;
+  __traits.max_shared_memory_per_multiprocessor = 100 * 1024;
+  __traits.max_blocks_per_multiprocessor        = 16;
+  __traits.max_warps_per_multiprocessor         = 48;
+  __traits.max_threads_per_multiprocessor =
+    __traits.max_warps_per_multiprocessor * detail::arch_common_traits::warp_size;
+  __traits.reserved_shared_memory_per_block = 1024;
+  __traits.max_shared_memory_per_block_optin =
+    __traits.max_shared_memory_per_multiprocessor - __traits.reserved_shared_memory_per_block;
+
+  __traits.cluster_supported  = false;
+  __traits.redux_intrinisic   = true;
+  __traits.elect_intrinsic    = false;
+  __traits.cp_async_supported = true;
+  __traits.tma_supported      = false;
+
+  return __traits;
+}();
+
+inline constexpr arch_traits_t sm_900_traits = []() constexpr {
+  arch_traits_t __traits{};
+  __traits.compute_capability_major             = 9;
+  __traits.compute_capability_minor             = 0;
+  __traits.compute_capability                   = 90;
+  __traits.max_shared_memory_per_multiprocessor = 228 * 1024;
+  __traits.max_blocks_per_multiprocessor        = 32;
+  __traits.max_warps_per_multiprocessor         = 64;
+  __traits.max_threads_per_multiprocessor =
+    __traits.max_warps_per_multiprocessor * detail::arch_common_traits::warp_size;
+  __traits.reserved_shared_memory_per_block = 1024;
+  __traits.max_shared_memory_per_block_optin =
+    __traits.max_shared_memory_per_multiprocessor - __traits.reserved_shared_memory_per_block;
+
+  __traits.cluster_supported  = true;
+  __traits.redux_intrinisic   = true;
+  __traits.elect_intrinsic    = true;
+  __traits.cp_async_supported = true;
+  __traits.tma_supported      = true;
+
+  return __traits;
+}();
+
+// TODO Should this be provided outside detail? Just using arch template seems better
+template <unsigned int __SmVersion>
+_CCCL_HOST_DEVICE inline constexpr arch_traits_t arch_traits()
+{
+  if constexpr (__SmVersion == 60)
+  {
+    return detail::sm_600_traits;
+  }
+  else if constexpr (__SmVersion == 70)
+  {
+    return detail::sm_700_traits;
+  }
+  else if constexpr (__SmVersion == 75)
+  {
+    return detail::sm_750_traits;
+  }
+  else if constexpr (__SmVersion == 80)
+  {
+    return detail::sm_800_traits;
+  }
+  else if constexpr (__SmVersion == 86)
+  {
+    return detail::sm_860_traits;
+  }
+  else
+  {
+    static_assert(__SmVersion == 90, "Unknown architecture");
+    return detail::sm_900_traits;
+  }
+}
+
+} // namespace detail
+
+//! @brief Retrieve architecture traits of the specified architecture
+//!
+//! @param __sm_version Compute capability in 10 * major + minor format for which the architecture traits are requested
+//!
+//! @throws cuda_error if the requested architecture is unknown
+_CCCL_HOST_DEVICE inline constexpr arch_traits_t arch_traits(unsigned int __sm_version)
```

Comment:
Suggestion: We already have several ways to spell the SM/PTX version. In CUB, we use `800` for Ampere. Here you are suggesting `80`. In libcu++ (see `libcudacxx/include/nv/target`) we actually have an enum [`sm_selector`](https://github.com/NVIDIA/cccl/blob/main/libcudacxx/include/nv/target#L85) with enumerator `sm_80` that would fit here. Maybe we can use one of the existing schemes instead of introducing a new one.

Reply:
I changed the format to 100 * major + 10 minor